### PR TITLE
Feat/#369 close app pre approval

### PR DESCRIPTION
**Pull request by ciaranschutte**
- add pre approval closed flow
- move entire form state up a level, so we can use `__refetchAllData` function in the top bar header too
- top progress bar always in closed state

### CODE REVIEW

File: components/pages/Applications/ApplicationForm/Forms/types.ts [hunk 1]
```diff
@@ -254,7 +254,8 @@ export type FormValidationAction = {
   value?: any;
 };
 
-interface FormValidationState_Base {
+export type RefetchData = (action?: Partial<FormValidationAction>) => void;
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/Forms/types.ts:257]: Perhaps I'm nitpicking, yet for the sake of uniformity, methinks this should be called RefetchDataFunction or something like that.
Thoughts?

File: components/pages/Applications/ApplicationForm/index.tsx [hunk 2]
```diff
@@ -52,17 +55,23 @@ const ApplicationForm = ({ appId = 'none', isAdmin = false }): ReactElement => {
...
       });
   }, [lastUpdated]);
 
-  return isLoading || (data && Object.values(data).length < 1) ? (
+  return isAppLoading || (data && Object.values(data).length < 1) ? (
     <Loader />
   ) : (
     <>
-      <ApplicationHeader data={data} />
+      <ApplicationHeader refetchAllData={formState.__refetchAllData} data={data} />
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/index.tsx:66]: Review note: ☝️ this right here is the gist of relocating `useFormValidation`.

Good call, @ciaranschutte. I had meant to elevate the refetching to this level.

File: components/pages/Applications/ApplicationForm/Forms/types.ts (hunk 1 above)
Comment (ciaranschutte) [reply components/pages/Applications/ApplicationForm/Forms/types.ts:257]: sounds great!

**Review by caravinci — APPROVED** [review]
🥾 squash this puppy!

**Pull request merged**